packages/core/src/actions/accounts/watchSigner.ts

```diff
@@ -16,8 +16,11 @@ export function watchSigner<TSigner extends Signer = Signer>(
   callback: WatchSignerCallback<TSigner>,
 ) {
   const client = getClient()
-  const handleChange = async () =>
-    callback(await fetchSigner<TSigner>({ chainId }))
+  const handleChange = async () => {
+    const signer = await fetchSigner<TSigner>({ chainId })
+    if (!getClient().connector) return callback(null)
```

Comment:
We need to get the client dynamically here, so can't use the client above (L18).

Reply:
Don't we already check that the connector exists inside [`fetchSigner`](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/actions/accounts/fetchSigner.ts#L15)?